zprice/src/serde.rs

```diff
@@ -0,0 +1,160 @@
+use ark_bls12_381::Bls12_381;
+use ark_serialize::{CanonicalDeserialize, CanonicalSerialize};
+use ark_std::{
+    boxed::Box,
+    io::{BufReader, Read, Write},
+    path::PathBuf,
+    print, println,
+    time::Instant,
+    vec::Vec,
+};
+use jf_plonk::prelude::*;
+
+const DEFAULT_UNIVERSAL_SRS_FILENAME: &str = "srs";
+const DEFAULT_PROVING_KEY_FILENAME: &str = "pk";
+const DEFAULT_VERIFICATION_KEY_FILENAME: &str = "vk";
+
+/// Create and store universal parameter in a file.
+pub fn store_srs(max_degree: usize, dest: Option<PathBuf>) {
+    let mut rng = rand::thread_rng();
+    let universal_param =
+        PlonkKzgSnark::<Bls12_381>::universal_setup(max_degree, &mut rng).unwrap();
+    let dest = match dest {
+        Some(dest) => dest,
+        None => default_path(DEFAULT_UNIVERSAL_SRS_FILENAME, "bin"),
+    };
+
+    let now = Instant::now();
+    print!(
+        "Storing universal parameter to: {} ...",
+        dest.to_str().unwrap()
+    );
+    store_data(&universal_param, dest);
```

Comment:
do you think we should also store a hash digest of the srs (which we will also put in the readme.md?)